File: src/main/java/com/ggardet/hospital/internal/entity/Hospital.java
package com.ggardet.hospital.internal.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Entity
@Data
@Table(name = "hospital", schema = "hospitals")
public class Hospital {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String address;
    
    @Column(name = "nb_max_services")
    private int nbMaxServices;
    
    @Column
    private String phoneNumber;
    
    @Column
    private String director;
}

File: src/main/kotlin/com/ggardet/hospital/internal/entity/Hospital.kt
package com.ggardet.hospital.internal.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "hospital", schema = "hospitals")
data class Hospital(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,
    
    @Column(nullable = false)
    var name: String,
    
    @Column(nullable = false)
    var address: String,
    
    @Column(name = "nb_max_services")
    var nbMaxServices: Int,
    
    @Column
    var phoneNumber: String? = null,
    
    @Column
    var director: String? = null
)